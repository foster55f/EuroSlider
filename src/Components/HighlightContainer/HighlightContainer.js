import React from 'react'
import HighlightCard from '../HighlightCard/HighlightCard';
import PropTypes from 'prop-types';
import { useSelector} from 'react-redux';



export const HighLightContainer = () => {
    const { games, favoriteGames } = useSelector(state => state);
    return (
        <div className= 'game-container'>
            {games.map((game, i) => {
            return (
                <HighlightCard
                key={i}
                id={game.competition.id} 
                title={game.title}
                image={game.thumbnail}
                url={game.url}
                embed={game.videos[0].embed}  
                isFavorited = {favoriteGames.includes(game)}    
            />
            )
        })}
        </div>
    )
}
export const mapStateToProps = state => ({
    games: state.games,
    displayGames: state.displayGames
  })
  
export default HighLightContainer

HighLightContainer.propTypes = {
    games: PropTypes.array.isRequired,
    displayGames: PropTypes.array.isRequired,
}