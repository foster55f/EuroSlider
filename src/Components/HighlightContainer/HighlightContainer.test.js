import React from 'react';
import { shallow } from 'enzyme';
import { HighLightContainer, mapStateToProps } from './HighlightContainer';

describe('HighlightContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<HighLightContainer
        games={[
          
      ]}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})

  describe('mapsStateToProps', () => {
    it('should return only game highlights from the store', () => {
        const mockState = {
            games: [],
            displayGames: [],
        };
        const expected = {
            games: [],
            displayGames: [],
        };
        const mappedProps = mapStateToProps(mockState);

        expect(mappedProps).toEqual(expected);
        });
    });