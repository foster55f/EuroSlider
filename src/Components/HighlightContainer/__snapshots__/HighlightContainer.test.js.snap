// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightContainer should match the snapshot 1`] = `
<div
  className="game-container"
/>
`;
